### DISCUSSION TITLE
Deploying a project using Vercel API

### DISCUSSION THREAD

**Discussion by rishavbharti** [vercel/community#2953] [Help] [answered]
### Summary

1. I'm attempting to deploy a Next.js project using the Vercel API, but I'm receiving an "Invalid JSON" response. I have provided a link to the CodeSandbox where this issue can be reproduced. Could you please assist me in understanding what might be causing this problem?

2. According to the [documentation](https://vercel.com/docs/rest-api/endpoints#create-a-new-project), it is stated that the `target` of an `environmentVariable` should be defined as `target: 'mport(/vercel/path0/utils/env-variable-util/types).EnvTarget.Productio'`. Can you please explain the path mentioned in target?

3. Can you confirm whether creating a project using the API also deploys it?

### Example

https://codesandbox.io/p/sandbox/vercel-api-demo-swq8p6

### Steps to Reproduce

1. **Create an Access Token** by following the [steps mentioned in the docs](https://vercel.com/docs/rest-api#creating-an-access-token).
2. In `app/api/route.ts`, **replace** `<---PUT YOUR TOKEN HERE--->` on Line 33 with the **token** generated above.
3. **Click on the "Deploy" button** in the preview window & observe the response.

**Comment by amyegan** [comment, marked as answer]
Hi @rishavbharti.

1. "Invalid JSON" response usually means an error page (HTML) is being returned when JSON was expected. I usually find it easiest to debug API calls using tools like Postman or Insomnia. I recommend giving that a try so you can get more information about the error.
2. It looks like there may be a glitch in our documentation. I believe the `target` should be the environment where you want the variable to be used. So it should be `production`, `preview`, or `development`. I'll check on why the page is showing the wrong target.
3. I think the "Create a new project" endpoint only creates the project. You can use the "Create a new deployment" endpoint to deploy your project via the API: https://vercel.com/docs/rest-api/endpoints#create-a-new-deployment
